tests/test_activate.py

```diff
@@ -1,652 +1,582 @@
+
+else:
+    # Only run these tests for commands that are installed.
+    shells = ['bash', 'zsh']
+    ps_var = "PS1"
+    var_format = "${var}"
+    binpath = "/bin/"  # mind the trailing slash.
+    source_setup = "source"
+    nul = '2>/dev/null'
+    set_var = 'export '
+    shell_suffix = ""
+    printps1 = '{echo} {var}'.format(echo=echo, var=var_format.format(var=ps_var))
+
+for shell in shells[:]:
+    try:
+        stdout, stderr = run_in('echo', shell)
+    except OSError:
+        pass
+    else:
+        if not stderr:
+            shells.append(shell)
+
+raw_ps = escape_curly(os.getenv(ps_var, ""))
+def print_ps1(env_dirs, number):
+    return u" ".join([u"({})".format(os.path.split(env_dirs[number])[-1]), raw_ps]).strip()
+
+CONDA_ENTRY_POINT = """\
 #!{syspath}/python
 import sys
 from conda.cli import main
 
 sys.exit(main())
 """
+printpath = '{echo} {var}'.format(echo=echo, var=var_format.format(var="PATH"))
+if sys.platform == "win32":
+    printdefaultenv = '{echo}.{var}'.format(echo=echo, var=var_format.format(var="CONDA_ACTIVE_ENV"))
+else:
+    printdefaultenv = '{echo} {var}'.format(echo=echo, var=var_format.format(var="CONDA_ACTIVE_ENV"))
 
 command_setup = """\
-export PATH="{ROOTPATH}"
-export PS1='$'
-export PYTHONPATH="{PYTHONPATH}"
-export CONDARC=' '
-cd {here}
-""".format(here=dirname(__file__), ROOTPATH=ROOTPATH, PYTHONPATH=PYTHONPATH)
-
-command_setup = command_setup + """
-mkdir -p {envs}/test1/bin
-mkdir -p {envs}/test2/bin
-"""
-
+set {ps_var}={raw_ps}
+set PYTHONPATH={PYTHONPATH}
+set CONDARC=
+""".format(here=dirname(__file__), PYTHONPATH=PYTHONPATH,
+           ps_var=ps_var, raw_ps=raw_ps)
+
+if platform.startswith("win"):
+    command_setup = "@echo off\n" + command_setup
+
+
+_format_vars = {
+    'nul': nul,
+    'printpath': printpath,
+    'printdefaultenv': printdefaultenv,
+    'printps1': printps1,
+    'set_var': set_var,
+    'source': source_setup,
+    'binpath': binpath,
+    'syspath': sys.prefix,
+    'shell_suffix': shell_suffix
+}
 
 @pytest.mark.slow
 def test_activate_test1():
     for shell in shells:
         with TemporaryDirectory(prefix='envs', dir=dirname(__file__)) as envs:
-            activate, deactivate, conda = _write_entry_points(envs)
             commands = (command_setup + """
-            source {activate} {envs}/test1
-            printf $PATH
-            """).format(envs=envs, activate=activate)
+            {source} {syspath}{binpath}activate{shell_suffix} "{env_dirs[0]}"
+            {printpath}
+            """).format(envs=envs, env_dirs=gen_test_env_paths(envs), **_format_vars)
 
             stdout, stderr = run_in(commands, shell)
-            assert stdout == envs + "/test1/bin:" + PATH
-            assert stderr == 'discarding {syspath} from PATH\nprepending {envs}/test1/bin to PATH\n'.format(envs=envs, syspath=syspath)
+            assert_equals(stderr, u'prepending {envpaths} to PATH'\
```

Comment:
I see your point, but I don't agree with it.  I don't think the root environment is necessarily on PATH.  I have been working to make that more achievable.  If paths to the activate scripts are provided absolutely, then there is no requirement, explicit or implicit, that root is active when starting a process.  Moreover, my thinking is that assuming no initial root env activation, we're in a better place.  When people want to work with conda and anaconda tools, they can start up a special "anaconda prompt" that drops them into the root environment (or directly into any other environment).  This is how Visual Studio does it.  People are still free to add the root environment to PATH.  If that's the case, then we never touch it - only prepend to PATH so earlier env gets found first.  This is less clean, because stuff not present in the prepended PATH may still get found in the later PATH.  I don't propose to fix this problem, but rather to provide tools towards the (cleaner) alternative.

Does this make sense to you, or do you see a better way?